grudge/op.py

```diff
@@ -819,6 +820,96 @@ def _apply_inverse_mass_operator(
     return DOFArray(actx, data=tuple(group_data))
 
 
+def _apply_inverse_mass_operator_quad(
+        dcoll: DiscretizationCollection, dd_out, dd_in, vec):
+    if not isinstance(vec, DOFArray):
+        return map_array_container(
+            partial(_apply_inverse_mass_operator_quad, dcoll, dd_out, dd_in), vec
+        )
+
+    from grudge.geometry import area_element
+
+    if dd_out != dd_in:
+        raise ValueError(
+            "Cannot compute inverse of a mass matrix mapping "
+            "between different element groups; inverse is not "
+            "guaranteed to be well-defined"
+        )
```

Comment:
Why have both `dd_in` and `dd_out`, if having them differ is not appropriate?

Reply:
I wondered that myself :).  Copied from the original `_apply_inverse_mass`: https://github.com/inducer/grudge/blob/23fbc05f50500130305aedcd970cdf6609882bb9/grudge/op.py#L798

Reply:
Fixed up in [45d6388](https://github.com/inducer/grudge/pull/354/commits/45d638884251489fca1c732879d9c9d687d7f019)